Blog post for v1.0.0 release (#2055)
woodpecker-ci/woodpecker: docs/blog/2023-07-28-release-v1.0.0/index.md

---
title: Presenting Woodpecker 1.0.0
description: Introducing Woodpecker 1.0.0 and its new features.
slug: release-v1.0.0
authors:
  - name: 6543
    title: Maintainer of Woodpecker
    url: https://github.com/6543
    image_url: https://github.com/6543.png
tags: [release, stable]
hide_table_of_contents: false
---

We are proud to present you Woodpecker v1.0.0.

> It took us quite some time, but now we are sure it's ready, and so you should really have a look at it.

Reviewer: ```suggestion
It took us quite some time, but now we are sure it's ready, and you should really have a look at it.
```

<!--truncate-->

> We've refactored a lot of code, so contributing to the codebase should be much easier.  

Reviewer: ```suggestion
We've refactored a lot of code, contributing to the codebase should be much easier.  
```

```suggestion
We've refactored a lot of code and contributing should be much easier now.  
```

> We've refactored a lot of code, so contributing to the codebase should be much easier.  

Reviewer: ```suggestion
We've refactored a lot of code and contributing to the codebase should be much easier now.
```

Author: well it could be better worded but I personally prefer the current wording over the suggested one

Reviewer: Thats totally fine

Furthermore, a ton of bugs where addressed and various enhancements introduced, along with some highly anticipated features.  
With Woodpecker v1.0.0, you can now substantially improve and streamline your code pipelines,
empowering you to automate and optimize workflows like never before.

## Some picked highlights:

### Add Support for Cron Jobs

Automate recurring tasks with ease using Woodpecker's new cron jobs feature.
Schedule pipelines to run at specified intervals or times, optimizing repetitive workflows.
[Read more](/docs/usage/cron)

### YAML Map Merge, Overrides, and Sequence Merge Support

With enhanced YAML support, managing complex configurations becomes a breeze.  
Merge maps, apply overrides, and sequence merging—all within your YAML files.

> This is providing greater flexibility and control over your pipelines.

Reviewer: ```suggestion
This is providing more flexibility and control over your pipelines.
```

[Read more](/docs/usage/advanced-yaml-syntax)

### Web-UI for Admins

Simplify administration tasks with Woodpecker's new Admin UI.  
Effortlessly manage user accounts, agents, and tasks, including adding new agents or pausing the task queue for maintenance.

![Image of admin queue view](./admin_queue_ui.png)

### Localize Web-UI

Embrace internationalization by changing your locale in the user settings.  
Interact with Woodpecker in the language of your choice, tailored to your preferences.  
If your language is not available or only partially translated, consider contributing to our [Weblate](https://translate.woodpecker-ci.org/engage/woodpecker-ci/).

### Add `evaluate` to `when` Filter

Enhance pipeline flexibility with the new "when evaluate" filter, enabling or disabling steps based on custom conditions.

> Tailor your workflows to respond dynamically to specific triggers and events.  

Reviewer: ```suggestion
Customize your workflows to dynamically respond to specific triggers and events.
```

Author: the `  ` at the end is intended to enforce a newline
